tests/phpunit/Unit/Service/SenderServiceTest.php

```diff
@@ -3,15 +3,20 @@
 namespace phpunit\Unit\Service;
 
 use Doctrine\ORM\EntityManagerInterface;
+use Lle\HermesBundle\Entity\Mail;
+use Lle\HermesBundle\Entity\Recipient;
+use Lle\HermesBundle\Entity\Template;
 use Lle\HermesBundle\Repository\MailRepository;
 use Lle\HermesBundle\Repository\RecipientRepository;
 use Lle\HermesBundle\Repository\UnsubscribeEmailRepository;
 use Lle\HermesBundle\Service\SenderService;
 use PHPUnit\Framework\TestCase;
 use Symfony\Component\DependencyInjection\ParameterBag\ParameterBagInterface;
 use Symfony\Component\Mailer\MailerInterface;
+use Symfony\Component\Routing\RequestContext;
 use Symfony\Component\Routing\RouterInterface;
 use Twig\Environment;
+use Twig\Loader\LoaderInterface;
```

Comment:
tu testSendAllMail ne contient aucun assert. Ca ne teste rien d'autre que l'implémentation des appels. Si on change l'implémentation on casse les tests mais on ne verifie pas que la fonction fait ce qu'on attend d'elle. Un test sert à tester un comportement, pas une implémentation

Reply:
there are 4 asserts in this test